### PR TITLE
Adding Windows Event to detections - Batch8

### PR DESCRIPTION
**Pull request by yaelrbergman**
Required items, please complete
   
   Change(s):
Added WindowsEvent source to the detections:
UserAccountAdd-Removed
UserAccountAddedToPrivlegeGroup_1h
FileHashEntity_SecurityEvent

   Reason for Change(s):
Added WindowsEvent data in addition to securityEvent

   Version Updated:
Yes

   Testing Completed:
Yes
   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

### CODE REVIEW

File: Detections/SecurityEvent/GroupCreatedAddedToPrivlegeGroup_1h.yaml [hunk 1]
```diff
@@ -35,15 +36,56 @@ query: |
...
+  ),
+  (
+  WindowsEvent 
+  // 4728 - A member was added to a security-enabled global group
+  // 4732 - A member was added to a security-enabled local group
+  // 4756 - A member was added to a security-enabled universal group  
+  | where EventID in ("4728", "4732", "4756")  and not(EventData has "S-1-5-32-555")
+  | extend SubjectUserSid = tostring(EventData.SubjectUserSid)
+  | extend Account = strcat(tostring(EventData.SubjectDomainName),"\\", tostring(EventData.SubjectUserName))
+  | extend AccountType=case(Account endswith "$", "Machine", iff(SubjectUserSid in ("S-1-5-18", "S-1-5-19", "S-1-5-20"), "Machine", iff(isempty(SubjectUserSid), "", "User")))
```
Comment (oshezaf) [inline Detections/SecurityEvent/GroupCreatedAddedToPrivlegeGroup_1h.yaml:49]: Sorry. Just noticed this now. The case/iff combination here is not needed and awkward. You probably wanted the below. Worth replacing backwards.

case 
    (Account endswith "$" or SubjectUserSid in ("S-1-5-18", "S-1-5-19", "S-1-5-20"), "Machine", 
     isempty(SubjectUserSid), "",
     "User")

File: Detections/SecurityEvent/GroupCreatedAddedToPrivlegeGroup_1h.yaml [hunk 2]
```diff
@@ -35,15 +36,56 @@ query: |
...
+  | extend MemberName = tostring(EventData.MemberName)
+  | where AccountType =~ "User" and MemberName == "-"
+  // Exclude Remote Desktop Users group: S-1-5-32-555
+  | extend TargetSid = tostring(EventData.TargetSid)
+  | where TargetSid !in ("S-1-5-32-555")
+  | where TargetSid matches regex WellKnownLocalSID or TargetSid matches regex WellKnownGroupSID
+  | extend TargetAccount = strcat(EventData.TargetDomainName,"\\", EventData.TargetUserName)
+  | extend SubjectAccount = strcat(EventData.SubjectDomainName,"\\", EventData.SubjectUserName)
```
Comment (oshezaf) [inline Detections/SecurityEvent/GroupCreatedAddedToPrivlegeGroup_1h.yaml:57]: You already calculated this as Account above.

File: Detections/SecurityEvent/UserAccountAdd-Removed.yaml [hunk 3]
```diff
@@ -25,26 +25,61 @@ query: |
...
   | extend  AccountAdded_GroupAddedTo_AddingAccount = strcat(AccountAdded, "||", GroupAddedTo, "||", AddingAccount )
-  | project AccountAdded_GroupAddedTo_AddingAccount, AccountAddedTime = TimeGenerated;
+  | project AccountAdded_GroupAddedTo_AddingAccount, AccountAddedTime = TimeGenerated
+  ),
+  (WindowsEvent
+  // Event ID related to member addition.
+  | where EventID in (4728, 4732,4756) 
+  | extend TargetSid = tostring(EventData.TargetSid)
+  | where TargetSid matches regex WellKnownLocalSID or TargetSid matches regex WellKnownGroupSID  
+  | parse EventData with * '"MemberName">' * '=' AccountAdded ",OU" *
```
Comment (oshezaf) [inline Detections/SecurityEvent/UserAccountAdd-Removed.yaml:44]: First, I would parse the relevant field, and not EventData as a whole. Secondly, there might be a bug here. Per [this](https://docs.microsoft.com/en-us/windows/security/threat-protection/auditing/event-4732) the field does not have ,OU in it.

File: Detections/SecurityEvent/UserAccountAdd-Removed.yaml [hunk 4]
```diff
@@ -25,26 +25,61 @@ query: |
...
+  | where isnotempty(AccountAdded)
+  | extend TargetUserName = tostring(EventData.TargetUserName)
+  | extend Account =  strcat(tostring(EventData.SubjectDomainName),"\\", tostring(EventData.SubjectUserName))
```
Comment (oshezaf) [inline Detections/SecurityEvent/UserAccountAdd-Removed.yaml:47]: Can assign directly to AddingAccount.

File: Detections/SecurityEvent/UserAccountAddedToPrivlegeGroup_1h.yaml [hunk 5]
```diff
@@ -39,6 +41,32 @@ query: |
   | extend SimpleMemberName = substring(MemberName, 3, indexof_regex(MemberName, @",OU|,CN") - 3)
   | project TimeGenerated, EventID, Activity, Computer, SimpleMemberName, MemberName, MemberSid, TargetUserName, TargetDomainName, TargetSid, UserPrincipalName, SubjectUserName, SubjectUserSid
   | extend timestamp = TimeGenerated, AccountCustomEntity = SimpleMemberName, HostCustomEntity = Computer
+  ),
+  (
+  WindowsEvent
```
Comment (oshezaf) [inline Detections/SecurityEvent/UserAccountAddedToPrivlegeGroup_1h.yaml:46]: Similar comments to the first detection.

**Review by oshezaf — CHANGES_REQUESTED** [review]

File: Detections/SecurityEvent/UserAccountAdd-Removed.yaml (hunk 3 above)
Comment (yaelrbergman) [reply Detections/SecurityEvent/UserAccountAdd-Removed.yaml]: I changed the parsing, what should I do with the potential bug?

Comment (oshezaf) [reply Detections/SecurityEvent/UserAccountAdd-Removed.yaml]: I will ask MSTIC

File: Detections/SecurityEvent/UserAccountAdd-Removed.yaml (hunk 4 above)
Comment (oshezaf) [reply Detections/SecurityEvent/UserAccountAdd-Removed.yaml]: I don't see the change here.

File: Detections/SecurityEvent/UserAccountAddedToPrivlegeGroup_1h.yaml [hunk 6]
```diff
@@ -39,6 +41,32 @@ query: |
...
+  ),
+  (
+  WindowsEvent 
+  // 4728 - A member was added to a security-enabled global group
+  // 4732 - A member was added to a security-enabled local group
+  // 4756 - A member was added to a security-enabled universal group
+  | where EventID in (4728, 4732, 4756)  and  not(EventData has "S-1-5-32-555")
+  | extend SubjectUserSid = tostring(EventData.SubjectUserSid)
+  | extend Account =  strcat(tostring(EventData.SubjectDomainName),"\\", tostring(EventData.SubjectUserName))
+  | extend AccountType=case(Account endswith "$", "Machine", iff(SubjectUserSid in ("S-1-5-18", "S-1-5-19", "S-1-5-20"), "Machine", iff(isempty(SubjectUserSid), "", "User")))
```
Comment (oshezaf) [inline Detections/SecurityEvent/UserAccountAddedToPrivlegeGroup_1h.yaml]: AccountType not corrected here.

**Review by oshezaf — CHANGES_REQUESTED** [review]

File: Detections/SecurityEvent/UserAccountAddedToPrivlegeGroup_1h.yaml (hunk 6 above)
Comment (yaelrbergman) [reply Detections/SecurityEvent/UserAccountAddedToPrivlegeGroup_1h.yaml]: @oshezaf what you see here is outdated, in this link you can see that the AccountType is updated.
https://github.com/Azure/Azure-Sentinel/blob/e9850fdb367aa3b0f2b35890482cf64e197fbd0f/Detections/SecurityEvent/UserAccountAddedToPrivlegeGroup_1h.yaml

**Review by oshezaf — APPROVED** [review]

**Pull request merged**